src/internal/task/etcd_service.go

```diff
@@ -118,15 +118,20 @@ func newEtcdDoer(namespaceEtcd *namespaceEtcd, group string, cache Cache) Doer {
 
 func (ed *etcdDoer) Do(ctx context.Context, inputChan chan *types.Any, cb CollectFunc) error {
 	return ed.withGroup(ctx, func(ctx context.Context, renewer *col.Renewer) error {
-		var eg errgroup.Group
 		prefix := path.Join(ed.group, uuid.NewWithoutDashes())
 		done := make(chan struct{})
 		var count int64
-		ctx, cancel := context.WithCancel(ctx)
 		defer func() {
-			cancel()
-			eg.Wait()
+			if _, err := col.NewSTM(ctx, ed.etcdClient, func(stm col.STM) error {
```

Comment:
I guess it doesn't matter given bryce's comment, but you could use the original `ctx` value here, rather than the error group context by doing `defer func(ctx context.Context){ ... }(ctx)`